clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -165,9 +167,10 @@ private void processApplicationEvents() {
 
         for (ApplicationEvent event : events) {
             try {
-                if (event instanceof CompletableEvent)
+                if (event instanceof CompletableEvent) {
                     applicationEventReaper.add((CompletableEvent<?>) event);
-
+                    maybeFailOnMetadataError(List.of((CompletableEvent<?>) event));
```

Comment:
Could you add a comment about why we're calling `maybeFailOnMetadataError()` here as well as at the end of `runOnce()`? I'm assuming this handles an important edge case, but I'm not sure I understand why from just looking at the code.